### PR TITLE
feat: [CU-860rmfjwh] create form to register + logout button

### PR DESCRIPTION
**Pull request by MASBERNATQ**
## Description

Setup [Supabase](https://supabase.com/), create a modal form to register user and add a button to logout the user.

## Changelog

> [!NOTE]
> Update your `.env.local` to add new environment variables :
>
> **NEXT_PUBLIC_SUPABASE_URL**
> **NEXT_PUBLIC_SUPABASE_ANON_KEY**

- Install _@supabase/supabase-js_ and setup supabase
- Add new alias _@auth/_ and remove global alias _@/_
- Create _useAuth_ hook to easily access to the user session
- Add button to logout session
- Create a _LoginModal_ component to register the player to supabase
- Create FieldText component
- Set guest player or logged player in the store
- Clean styles and improve some models

## Related ticket

This pull request is related to ClickUp ticket : **[CU-860rmfjwh](https://app.clickup.com/t/860rmfjwh)**.

## Checklist

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have named the title of my pull request while respecting the naming convention:

```
<type>(<scope>): [<ticket id>] <short summary>
  │       │            │              │
  │       │            │              └─ Summary in present tense. Not capitalized. No period at the end.
  │       │            │
  │       │            └─ ClickUp ticket id (optional)
  │       │
  │       └─ Commit Scope (optional)
  │
  └─ Commit Type: build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test
```

### CODE REVIEW

**Comment by FlavianEng** [comment]
Task linked: [CU-860rmfjwh Create form to register](https://app.clickup.com/t/860rmfjwh)

**Comment by FlavianEng** [comment]
> ## Description
> Setup [Supabase](https://supabase.com/), create a modal form to register user and add a button to logout the user.
> 
> ## Changelog
> Note Update your `env.local` to add new environment variables :
> 
> **NEXT_PUBLIC_SUPABASE_URL** **NEXT_PUBLIC_SUPABASE_ANON_KEY**
> 
> * Install _@supabase/supabase-js_ and setup supabase
> * Add new alias _@auth/_ and remove global alias _@/_
> * Create _useAuth_ hook to easily access to the user session
> * Add button to logout session
> * Create a _LoginModal_ component to register the player to supabase
> * Create FieldText component
> * Set guest player or logged player in the store
> * Clean styles and improve some models
> 
> ## Related ticket
> This pull request is related to ClickUp ticket : **[CU-860rmfjwh](https://app.clickup.com/t/860rmfjwh)**.
> 
> ## Checklist
> * [x]  I have performed a self-review of my code
> * [x]  I have commented my code, particularly in hard-to-understand areas
> * [x]  I have made corresponding changes to the documentation
> * [x]  I have added tests that prove my fix is effective or that my feature works
> * [x]  I have named the title of my pull request while respecting the naming convention:
> 
> ```
> <type>(<scope>): [<ticket id>] <short summary>
>   │       │            │              │
>   │       │            │              └─ Summary in present tense. Not capitalized. No period at the end.
>   │       │            │
>   │       │            └─ ClickUp ticket id (optional)
>   │       │
>   │       └─ Commit Scope (optional)
>   │
>   └─ Commit Type: build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test
> ```

Can't you push your `.env.local instead` ?

File: src/shared/_hooks/useClientSideValue.tsx [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
-function useClientSideValue<T>(randomValue: T): T {
-  const [value, setValue] = useState<T>(randomValue);
+function useClientSideValue<T>(v: T): T {
+  const [value, setValue] = useState<T>(v);
 
   useEffect(() => {
-    setValue(randomValue);
-  }, [randomValue]);
+    setValue(v);
+  }, [v]);
```
Comment (FlavianEng) [inline src/shared/_hooks/useClientSideValue.tsx:15]: Can you rename this variable? I don't think `v` is a meaningful variable name

Comment (MASBERNATQ) [reply src/shared/_hooks/useClientSideValue.tsx:15]: :heavy_check_mark:

**Comment by MASBERNATQ** [comment]
@FlavianEng je préfères ne pas push mon `.env.local` pour ne pas mettre à disposition des variables secrètes même si elles concernent l'environnement de développement comme Supabase.

**Review by FlavianEng — DISMISSED** [review]

**Review by FlavianEng — APPROVED** [review]

**Pull request merged**